### PR TITLE
Added discussion of make variables, shell variables, and quoting

### PR DESCRIPTION
**Pull request by bernhold**
Addresses issue #75

This is for instructor-training.

### CODE REVIEW

**Comment by bernhold** [comment]
Minor update to implement @KamilSJaron suggestion to relate GNU make's 'quoting' with the unix shell's 'escape' concept.

File: _extras/discuss.md [hunk 1]
```diff
@@ -76,6 +76,89 @@ The configuration file then simply contains the default values for the
...
+  there might be ambiguity in interpreting variable names, or for
+  certain pattern substitution operations.  Since there are only
+  certain situations in which the unix shell requires this syntax,
+  instead of the more common `$NAME`, it is not familiar to many users.
+
+* Make does variable substitution on actions before they are passed to
+  the shell for execution.  That means that anything that looks like a
+  variable to make will get replaced with the appropriate value.  (In
+  make, an uninitialized variable has a null value.)  To protect
+  variable you intend to be interpreted by the shell rather than make,
```
Comment (gcapes) [inline _extras/discuss.md]: Missing 'a'
> To protect **a** variable

File: _extras/discuss.md [hunk 2]
```diff
@@ -76,6 +76,89 @@ The configuration file then simply contains the default values for the
...
+> the action that would be passed to the shell to execute would be:
+> 
+> ~~~
+> for book in abyss isles; do python plotcount.py ook.dat ook.png; done
+> ~~~
+> {: .bash}
+> 
+> Notice that make substituted `$(BOOKS)`, as expected, but it also
+> substituted `$book`, even though we intended it to be a shell variable.
+> Moreover, because it didn't use `$(NAME)` (or `${NAME}`) syntax, make
```
Comment (gcapes) [inline _extras/discuss.md]: because **we** didn't use use `$(NAME)` (or `${NAME}`) syntax,

**Comment by bernhold** [comment]
Updated to respond to @gcapes requests.

**Comment by gcapes** [comment]
Thanks!

**Pull request merged**